src/lib.rs

```diff
@@ -338,6 +594,19 @@ macro_rules! up {
             },
         }
     );
+    ($e:expr, $($key:expr, $value:expr), *) => (
```

Comment:
What would you think of bikeshedding the syntax here a bit? Since we're already using a macro, we might as well make the intention clearer with explicit key/value syntax. 

I'm thinking of something like `throw_new!("Error", code=78, application="rust_core")`, but there are many different possibilities.

Would apply to other macros as well.